src/testsuite/arith128_test_i64.c

```diff
@@ -1094,6 +1272,758 @@ test_muloud (void)
+
+  i = (vui64_t){ 0x100000000, 0x100000000 };
+  j = (vui64_t){ 0x100000000, 0x200000000 };
+  e = (vui64_t){ 0x1, 0x2 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t){ 0x200000000, 0x200000000 };
+  j = (vui64_t){ 0x300000000, 0x400000000 };
+  e = (vui64_t){ 0x6, 0x8 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t){ 1000000000000000000, 1000000000000000000 };
+  j = (vui64_t){ 19, 18447 };
+  e = (vui64_t){ 1, 1000 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t){ 1000000000000000000, 1000000000000000000 };
+  j = (vui64_t){ 18446745, 18446744074 };
+  e = (vui64_t){ 1000000, 1000000000 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t){ __UINT64_MAX__, __UINT64_MAX__ };
+  j = (vui64_t){ 19, 18447 };
+  e = (vui64_t){ 18, 18446 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t){ __UINT64_MAX__, __UINT64_MAX__ };
+  j = (vui64_t){ 18446745, 18446744074 };
+  e = (vui64_t){ 18446744, 18446744073 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t){ __UINT64_MAX__, __UINT64_MAX__ };
+  j = (vui64_t){ __UINT64_MAX__, __UINT64_MAX__ };
+  e = (vui64_t){ __UINT64_MAX__-1, __UINT64_MAX__-1 };
+  k = vec_mulhud (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("mulhud  ( ", i);
+  print_v2int64 ("         ,", j);
+  print_v2int64 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_mulhud:", (vui128_t)k, (vui128_t) e);
+
+  return (rc);
+}
+
+int
+test_muludm (void)
+{
+  vui64_t i, j;
+  vui64_t k, e;
+  int rc = 0;
+
+  printf ("\ntest_muludm Vector Multiply Unsigned doubleword Modulo\n");
+  i = (vui64_t){1, 2};
+  j = (vui64_t){101, 102};
+  e = (vui64_t){101, 204};
+  k = vec_muludm(i, j);
```

Comment:
space after function.  _global_, as this occurs frequently.

Reply:
OK